docs/shared/workspace/structure/dependency-graph.md

```diff
@@ -0,0 +1,49 @@
+# Analyzing & Visualizing Workspaces
+
+To be able to support the monorepo-style development, the tools must know how different projects in your workspace depend on each other. Nx uses advanced code analysis to construct this dependency graph. And it gives you a way to explore it:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/cMZ-ReC-jWU" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+## How the Dependency Graph is Built
+
+Nx creates a graph of all the dependencies between projects in your workspace using two sources of information:
+
+1. Typescript `import` statements referencing a particular project's path alias
+
+   For instance, if a file in `my-app` has this code:
+
+   ```typescript
+   import { something } from '@myorg/awesome-library';
+   ```
+
+   Then `my-app` depends on `awesome-library`
+
+2. Manually created `implicitDependencies` in the `nx.json` file. [Full `implicitDependencies` documentation](/{framework}/workspace/configuration#implicit-dependencies)
+
+   If your `nx.json` has this content:
+
+   ```json
+   {
+     "projects": {
+       "my-app": {
+         "tags": [],
+         "implicitDependencies": ["my-api"]
+       }
+     }
+   }
+   ```
+
+   Then `my-app` depends on `my-api`
+
+## Circular Dependencies
+
+A circular dependency is when a project transitively depends on itself. This can cause problems in the design of your software and also makes Nx's affected algorithm less effective. The lint rule, `nx-enforce-module-boundaries`, warns you if any circular dependencies are created and ensures that any `import` statements going across projects only `import` from the defined public apis in a project's root `index.ts` file.
```

Comment:
One thing to add is that the lint rule produces an error when running the lint checks